src/Microsoft.CodeQuality.Analyzers/CSharp/QualityGuidelines/CSharpRemoveEmptyFinalizersAnalyzer.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using System.Linq;
+using Analyzer.Utilities;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeQuality.Analyzers.QualityGuidelines;
+
+namespace Microsoft.CodeQuality.CSharp.Analyzers.QualityGuidelines
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    internal class CSharpRemoveEmptyFinalizersAnalyzer : AbstractRemoveEmptyFinalizersAnalyzer
+    {
+        protected override bool IsEmptyFinalizer(SyntaxNode methodBody, SymbolAnalysisContext analysisContext)
+        {
+            var destructorDeclaration = (DestructorDeclarationSyntax)methodBody;
```

Comment:
We can share large amount of code between C# and VB implementations. If we are planning to revert this to IOperation analyzer soon, then I am fine with code duplication. @jinujoseph are we going to revert this change once underlying IOperation issue is fixed?